haystack/nodes/prompt/prompt_node.py

```diff
@@ -708,18 +723,11 @@ def prompt(self, prompt_template: Optional[Union[str, PromptTemplate]], *args, *
         results = []
         # we pop the prompt_collector kwarg to avoid passing it to the model
         prompt_collector: List[Union[str, List[Dict[str, str]]]] = kwargs.pop("prompt_collector", [])
-        if isinstance(prompt_template, str) and not self.is_supported_template(prompt_template):
-            raise ValueError(
-                f"{prompt_template} not supported, please select one of: {self.get_prompt_template_names()} "
-                f"or pass a PromptTemplate instance for prompting."
-            )
 
         # kwargs override model kwargs
         kwargs = {**self._prepare_model_kwargs(), **kwargs}
-        prompt_template_used = prompt_template or self.default_prompt_template
-        if prompt_template_used:
-            template_to_fill = self.get_prompt_template(prompt_template_used)
-
+        template_to_fill = self.get_prompt_template(prompt_template)
+        if template_to_fill:
```

Comment:
No need for extra logic here. `get_prompt_template` is fully responsible for resolving the prompt template.